### ISSUE TITLE
config about pretrain tiny imagenet

### ISSUE THREAD

**Issue by 1ranGuan** [Hansong-Zhang/DANCE#1]
Thank you for your excellent work. When we run your code, the model fails to converge when pre-training tiny imagenet. The config and training results are as follows. I would like to ask for the correct config settings so that we can reproduce the performance in your paper. Thank you!
DSA strategy:  color_crop_flip_scale_rotate
Pretrain models save dir: /data/dd_dance_tiny
Namespace(dataset='tinyimagenet', data_dir='/data/tiny_imagenet', imagenet_dir='/ssd_data/imagenet/', nclass=200, dseed=0, size=64, phase=-1, nclass_sub=-1, load_memory=True, net_type='convnet', norm_type='instance', depth=3, width=1.0, pretrain_epochs=60, batch_real=64, batch_size=64, lr=0.01, momentum=0.9, weight_decay=0.0005, seed=0, pretrain_amount=20, mixup='cut', beta=1.0, mix_p=0.5, rrc=True, dsa=True, dsa_strategy='color_crop_flip_scale_rotate', aug_type='color_crop_cutout', verbose=False, workers=8, ipc=-1, reproduce=True, bottleneck=True, data_name='tinyimagenet', pretrain_dir='/data/dd_dance_tiny', niter=10000, nch=3, augment=False)
Augmentataion Matching:  color_crop_cutout
Augmentataion Net update:  color_crop
Start training 20 models for 60 epochs
<Pretraining  0-th model>...[Epoch  0] Train acc: 3.9 (loss: 5.023), Val acc: 1.3
<Pretraining  0-th model>...[Epoch  1] Train acc: 9.8 (loss: 4.492), Val acc: 1.2
<Pretraining  0-th model>...[Epoch  2] Train acc: 13.9 (loss: 4.225), Val acc: 0.4
<Pretraining  0-th model>...[Epoch  3] Train acc: 16.7 (loss: 4.046), Val acc: 1.0
<Pretraining  0-th model>...[Epoch  4] Train acc: 18.7 (loss: 3.937), Val acc: 0.8
<Pretraining  0-th model>...[Epoch  5] Train acc: 20.1 (loss: 3.848), Val acc: 0.9
<Pretraining  0-th model>...[Epoch  6] Train acc: 21.1 (loss: 3.789), Val acc: 0.9

**Comment by Hansong-Zhang** [comment]
Hi Yiran,

Sorry for the delayed reply. The model utilized to distill tinyimagenet is ConvNet-4, while your config is ConvNet-3, maybe you can check this and see if it works. BTW, is there any question when you pretraining on other datasets like CIFAR? If not, maybe you can check if your dataset is defined correctly (e.g. datapath, size, etc).

Warm regards,
Hansong

**Comment by Hansong-Zhang** [comment]
> Thank you for your excellent work. When we run your code, the model fails to converge when pre-training tiny imagenet. The config and training results are as follows. I would like to ask for the correct config settings so that we can reproduce the performance in your paper. Thank you! DSA strategy: color_crop_flip_scale_rotate Pretrain models save dir: /data/dd_dance_tiny Namespace(dataset='tinyimagenet', data_dir='/data/tiny_imagenet', imagenet_dir='/ssd_data/imagenet/', nclass=200, dseed=0, size=64, phase=-1, nclass_sub=-1, load_memory=True, net_type='convnet', norm_type='instance', depth=3, width=1.0, pretrain_epochs=60, batch_real=64, batch_size=64, lr=0.01, momentum=0.9, weight_decay=0.0005, seed=0, pretrain_amount=20, mixup='cut', beta=1.0, mix_p=0.5, rrc=True, dsa=True, dsa_strategy='color_crop_flip_scale_rotate', aug_type='color_crop_cutout', verbose=False, workers=8, ipc=-1, reproduce=True, bottleneck=True, data_name='tinyimagenet', pretrain_dir='/data/dd_dance_tiny', niter=10000, nch=3, augment=False) Augmentataion Matching: color_crop_cutout Augmentataion Net update: color_crop Start training 20 models for 60 epochs <Pretraining 0-th model>...[Epoch 0] Train acc: 3.9 (loss: 5.023), Val acc: 1.3 <Pretraining 0-th model>...[Epoch 1] Train acc: 9.8 (loss: 4.492), Val acc: 1.2 <Pretraining 0-th model>...[Epoch 2] Train acc: 13.9 (loss: 4.225), Val acc: 0.4 <Pretraining 0-th model>...[Epoch 3] Train acc: 16.7 (loss: 4.046), Val acc: 1.0 <Pretraining 0-th model>...[Epoch 4] Train acc: 18.7 (loss: 3.937), Val acc: 0.8 <Pretraining 0-th model>...[Epoch 5] Train acc: 20.1 (loss: 3.848), Val acc: 0.9 <Pretraining 0-th model>...[Epoch 6] Train acc: 21.1 (loss: 3.789), Val acc: 0.9

Hi Yiran,

Sorry for the delayed reply. The model utilized to distill tinyimagenet is ConvNet-4, while your config is ConvNet-3, maybe you can check this and see if it works. BTW, is there any question when you pretraining on other datasets like CIFAR? If not, maybe you can check if your dataset is defined correctly (e.g. datapath, size, etc).

Warm regards,
Hansong

**Comment by fencingx** [comment]
Hello Hansong and Yiran,
   I have the same problem, how did you solve it? Thanks
